### ISSUE TITLE
feature: Allow for focusing main code window while a response is being streamed

### ISSUE THREAD

**Issue by johnsaigle** [yetone/avante.nvim#973]
### Feature request

Currently when using Avante Ask the streaming response will "steal" focus in the main window as the model's text is populated in the buffer.

My preference would be to have the ability to navigate in the main code window while the response streams on the side. Right now this isn't possible because the cursor and focus jumps to the response buffer.

#### Example 1
While using Avante Ask, the response is streamed and each new bit of text causes the cursor to move away from the main window into the Avante Ask window.

#### Example 2
When using Avante Edit, the small prompt window steals focus. Instead of navigating the code, new keystrokes are appended to the prompt window even after the prompt has been sent.

**Comment by albankurti** [comment]
fully agreed, there is very little utility to having the attention on the chat as a response is generated, even worse - you can't step out of the chat it keeps stealing the focus :)

**Comment by albankurti** [comment]
Created [PR 982](https://github.com/yetone/avante.nvim/pull/982) which should allow you to have this behavior. 

The config should be updates this way:
```
  {
    "yetone/avante.nvim",
    ...
    opts = {
      ...
      behaviour = {
        auto_focus_sidebar = false,
      },
      ...
  },
```

If you would like to test it and provide feedback it would be nice :) Just switch the .local/share/nvim/lazy/avante.nvim to the branch in the PR, or wherever your packages are located

**Comment by braveltd** [comment]
great work guys!

**Comment by JunYang-tes** [comment]
hi @albankurti , thanks for your contribution. 
by setting auto_focus_sidebar to true, the input area won't focused when I open the sidebar.I think this is a little inconvenient because we open sidebar  usually means we need to input something. I personally prefer have a option called auto_focus_result_window to control if the result window will be focused or not. what do you think?

**Comment by albankurti** [comment]
@JunYang-tes I agree, further I noticed that insert mode is not working well during generation. I will attempt to work on these two things.
